doc/source/terrain.md

```diff
@@ -10,8 +24,87 @@ and tested for consistency against [gdaldem](https://gdal.org/programs/gdaldem.h
 ## Quick use
 
 Terrain attribute methods can either be called directly from a {class}`~xdem.DEM` (e.g., {func}`xdem.DEM.slope`) or
-through the {class}`~xdem.terrain` module which allows array input.
+through the {class}`~xdem.terrain` module (e.g., {func}`xdem.terrain.slope`) which allows a NumPy array input but
+requires additional metadata.
 
+```{code-cell} ipython3
+:tags: [remove-cell]
+
+# To get a good resolution for displayed figures
+from matplotlib import pyplot
+pyplot.rcParams['figure.dpi'] = 600
+pyplot.rcParams['savefig.dpi'] = 600
+```
+
+```{code-cell} ipython3
+:tags: [hide-cell]
+:mystnb:
+:  code_prompt_show: "Show the opening of example files."
+:  code_prompt_hide: "Hide the opening of example files."
+
+import xdem
+
+# Open a DEM from a filename on disk
+filename_dem = xdem.examples.get_path("longyearbyen_ref_dem")
+dem = xdem.DEM(filename_dem)
+```
+
+```{code-cell} ipython3
+# Slope from DEM method
+slope = dem.slope()
+# Or from terrain module using an array input
+slope = xdem.terrain.slope(dem.data, resolution=dem.res)
+```
+
+If computational performance is key, xDEM can rely on [RichDEM](https://richdem.readthedocs.io/) by specifying
+`use_richdem=True` for speed-up of specific supported attributes (slope, aspect, curvature).
+
+## Summary of supported methods
+
+```{list-table}
+   :widths: 1 1 1
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Attribute
```

Comment:
Very useful table! Especially the unit column.
I am also thinking, should we raise a warning when trying to call many of these functions from geographic coordinates? Because it will return unrealistic values (e.g. slope).
Also for the hillshade, we could detect this and add the correction factor.

Reply:
Added issue #634.

Reply:
Perfect. I got exactly this issue the other day.